padl/transforms.py

```diff
@@ -861,10 +907,21 @@ def infer_apply(self, inputs):
         :param inputs: The input.
         """
         self.pd_forward_device_check()
+        in_args = inputs
+        global _pd_trace
+        _pd_trace.clear()
         inputs = self.pd_preprocess.pd_call_transform(inputs, mode='infer')
         inputs = _move_to_device(inputs, self.pd_device)
-        inputs = self.pd_forward.pd_call_transform(inputs, mode='infer')
-        inputs = self.pd_postprocess.pd_call_transform(inputs, mode='infer')
+        try:
+            inputs = self.pd_forward.pd_call_transform(inputs, mode='infer')
+        except Exception as err:
+            self._pd_trace_error(self._pd_get_error_idx('forward'), in_args)
+            raise err
```

Comment:
Why do we need to have the `try` and `except` to wrap the `pd_call_transform`, and also be present in the `_pd_call_transform`, which is called by `pd_call_transform`?

Reply:
Because in `pd_call_transform` we are tracing the error of `self.pd_preprocess`, `self.pd_forward` or `self.pd_postprocess`, which are the transforms that apply `pd_call_transform`. In this `try` we are adding one more element to the trace where we inspect the entire transform (`pd_preprocess`, `pd_forward` + `pd_postprocess`).